### ISSUE TITLE
userPing incorrectly uses ping on BSD for ipv6 addresses

### ISSUE THREAD

**Issue by aschleck** [tailscale/tailscale#8225]
### What is the issue?

Thank you for this amazing software!

I am trying to get subnet routing over ipv6 working, and have been debugging using ping6 from my local Mac laptop. After going in circles for a while I noticed that my pfSense subnet router is showing:

```
May 27 17:00:35 | tailscaled | 37950 | 2023/05/27 17:00:35 exec ping of fd05:b97c:a00:0:75b0:c184:d219:3f57 failed in 260.14298ms: exit status 68
```

If on the subnet router machine I do the equivalent of what [userPing](https://github.com/tailscale/tailscale/blob/main/wgengine/netstack/netstack.go#L637) does, `ping fd05:b97c:a00:0:75b0:c184:d219:3f57` then I see "No address associated with that name" and get exit code 68 which matches.

If I do `ping6 fd05:b97c:a00:0:75b0:c184:d219:3f57` then it works, so I think Tailscale needs to also use ping6 in the IPv6 case under BSD (and Darwin is likely broken for the same reason.)

### Steps to reproduce

* Install Tailscale as an IPv6 ULA subnet router on pfSense
* Try pinging a host in the IPv6 subnet from another computer in the Tailnet

### OS

Other

### OS version

pfSense 2.6.0

### Tailscale version

1.36.0

### Bug report

BUG-26fc0ab1c860397fa0017ffd149fc5d962e4c05b88fd2ad1d36caaf5bfff3fa4-20230528000629Z-fb8484e269f67a5c

**Comment by DentonGentry** [comment]
Could I ask what version of FreeBSD is shown in `uname -a`?

When I run ping on a FreeBSD 13.1 system, it figures out that the destination address is IPv6:
```
# ping fd05:b97c:a00:0:75b0:c184:d219:3f57
PING6(56=40+8+8 bytes) 2602:24c:b8d:a2f0:: --> fd05:b97c:a00:0:75b0:c184:d219:3f57
^C
--- fd05:b97c:a00:0:75b0:c184:d219:3f57 ping6 statistics ---
3 packets transmitted, 0 packets received, 100.0% packet loss
# ping6 fd05:b97c:a00:0:75b0:c184:d219:3f57
PING6(56=40+8+8 bytes) 2602:24c:b8d:a2f0:: --> fd05:b97c:a00:0:75b0:c184:d219:3f57
^C
--- fd05:b97c:a00:0:75b0:c184:d219:3f57 ping6 statistics ---
3 packets transmitted, 0 packets received, 100.0% packet loss
```

It cannot actually ping anything because I'm using the IPv6 address from your example, which doesn't exist here, but it tried.

**Comment by DentonGentry** [comment]
Though there may well be a problem in this case with `ping` vs `ping6`, I think there is another problem as well. In telemetry somewhat before the bugreport, we see:
```
2023-05-27 23:10:12.378592709 +0000 UTC: Drop: ICMPv6{[fd7a:115c:a1e0:ab12:q:r:s:t]:0 > [fde5:a54c:a00:0:75b0:c184:d219:3f57]:0} 56 no rules matched
2023-05-27 23:10:22.402538543 +0000 UTC: Drop: ICMPv6{[fd7a:115c:a1e0:ab12:q:r:s:t]:0 > [fde5:a54c:a00:0:75b0:c184:d219:3f57]:0} 56 no rules matched
```

The IPv6 subnet will need to be allowed in the ACLs. Unfortunately in attempting to do so you'll run into another problem, that the ACL parser in the coordination server doesn't handle IPv6 routes directly in an ACL entry. You'll need to use an entry in the `Hosts` block:
```
	"Hosts": {
		"ip6subnet":  "fd05:b97c:a00:0::/64",
	},
	"ACLs": [
		{
			"action": "accept",
			"users":  ["*"],
			"ports":  ["ip6subnet:*"],
		},
```

**Comment by DentonGentry** [comment]
`tailscaled` can function as a subnet router in one of two ways:
- kernel forwarding, used on Linux when a `/dev/net/tun` device is available
- netstack (aka "hybrid") forwarding entirely in userspace, used on *BSD, macOS, Windows, and Linux with `--tun=userspace-networking`

In netstack forwarding mode it handles ping by exec of the ping command. That happens in https://github.com/tailscale/tailscale/blob/c0b4a54146e23e0600f43c1a39af956edf62ab54/wgengine/netstack/netstack.go#L614

FreeBSD and Linux are both handled by the default case of that routine. I suppose if we need to, GOOS="freebsd" could differentiate if the destination address is IPv6 and invoke `ping6` instead.

**Comment by aschleck** [comment]
Wow thank you Denton for such a quick reply!

I should have clarified that I was in the end able to get routing working. I just hadwasted a tremendous amount of time because I was trying to use `ping` to check connectivity, and Tailscale was dropping the pings due to this bug. After starting an HTTP server on the remote computer (and when I finally remembered to disable the firewall on that computer) the connection worked fine. I had added `*:*` to the ACL, but thank you so much for the `ip6subnet` example as that's exactly what I was trying to accomplish and had no idea how to do.

Interesting regarding it working on 13.1! I guess pfSense is a little behind at 12.3. My Macs (on Ventura, and another on Monterey) also do not support IPv6 via `ping` and require `ping6` so that's why I think Darwin is affected too. 

```
$ uname -a
FreeBSD <myhostname> 12.3-STABLE FreeBSD 12.3-STABLE RELENG_2_6_0-n226742-1285d6d205f pfSense  amd64
```
